frontend/src/utils/variants.ts

```diff
@@ -18,7 +18,8 @@ export interface Variant {
 export const getVariantValue = <T = string>(
     variant: Variant | undefined
 ): T | undefined => {
-    if (variant?.payload !== undefined) {
+    if (variant?.enabled) {
+        if (!variant.payload) return variant.name as T;
```

Comment:
That was the intention, yes. In case you don't specify a payload, the only thing you can use to identify the variant and runtime control, effectively its value in the perspective of this helper, turns out to be its name.

Reply:
In case it helps visualizing what I mean in our UI:
<img width="1207" alt="image" src="https://github.com/Unleash/unleash/assets/14320932/c778dbcb-f816-440e-bfda-be52ab5e5153">

This means that we could easily do something like:

```TS
const variant = getVariantValue(uiConfig.flags.ourVariant)

if (variant === 'variant-A') {
  // Do cool thing
}
```